Assets/Mirage/ObjectPolling/PooledObject.cs

```diff
@@ -0,0 +1,262 @@
+        /// <returns></returns>
+        public NetworkIdentity NetworkSpawn(Vector3 position, Quaternion rotation)
+        {
+            while (true)
+            {
+                NetworkIdentity spawnedObject;
+
+                switch (_inactiveGameObjects.Count)
+                {
+                    case 0:
+                        spawnedObject = Object.Instantiate(_networkObjectPrefabToPool, position, rotation);
+
+                        PooledObjectIDs.Add(spawnedObject.gameObject.GetInstanceID());
+
+                        return spawnedObject;
+                    default:
+                        spawnedObject = _inactiveGameObjects.Dequeue().NetId;
+
+                        if (spawnedObject is null)
+                            continue;
+
+                        break;
+                }
+
+                spawnedObject.transform.SetParent(null);
+                spawnedObject.transform.position = position;
+                spawnedObject.transform.rotation = rotation;
+
+                return spawnedObject;
+            }
+        }
+
+        /// <summary>
+        ///     Method to spawn network object from our inactive pool.
+        /// </summary>
+        /// <param name="parentTransform">The transform to parent object to.</param>
+        /// <param name="worldStayPosition">Unity specific <see cref="Instantiate" /></param>
+        /// <returns></returns>
+        public NetworkIdentity NetworkSpawn(Transform parentTransform, bool worldStayPosition)
+        {
+            while (true)
+            {
+                NetworkIdentity spawnedObject;
+
+                if (_inactiveGameObjects.Count > 0)
+                {
+                    spawnedObject = _inactiveGameObjects.Dequeue().NetId;
+
+                    if (spawnedObject is null)
+                        continue;
+
+                    spawnedObject.transform.SetParent(parentTransform, worldStayPosition);
+
+                    return spawnedObject;
+                }
+
+                spawnedObject = Object.Instantiate(_networkObjectPrefabToPool, parentTransform, worldStayPosition);
+
+                PooledObjectIDs.Add(spawnedObject.gameObject.GetInstanceID());
+
+                return spawnedObject;
+            }
+        }
+
+        /// <summary>
+        ///     Method to return object back to the pool for re-usage.
+        /// </summary>
+        /// <param name="obj">The object we want to return back to the pool.</param>
+        internal void Despawn(NetworkIdentity obj)
+        {
+            if (!obj.gameObject.activeInHierarchy) return;
+
+            obj.gameObject.SetActive(false);
+
+            obj.transform.SetParent(_parentBackObject.transform);
+
+            obj.transform.position = Vector3.zero;
+            obj.transform.rotation = Quaternion.identity;
+
+            var recycle = new PooledObject<T>();
+
+            _inactiveGameObjects.Enqueue(recycle);
```

Comment:
you are not setting any field on this new object so `obj` is never added back to the pool

Reply:
Your not adding the object to the pool. Your adding back the struct data. To inform that we have now an open slot to reuse. This is better and alot fast then dumping the gameobject back. The gameobject itself turns off with active that's the idea.

This is much faster and allows to know we have a new object in pool to use.